### PR TITLE
fix: testing issues

### PR DESCRIPTION
**Pull request by albertfolch-redeemeum**
- feat: add code & assets from uniswap to show erc20 tokens and native networks images in tokenGated
- fix: tokenURI function
- fix: show erc721 and erc1155 images in tokenGated
- fix: casing inconsistencies
- fix: erc721 and erc1155 ID not breaking if it's too long
- feat: change date format in details part to D MMM. YYYY
- feat: use useQuery to fetch exchange policy instead of useEffect
- fix: ipfs parsing (getIpfsGatewayUrl) to support ipfs links that have more slashes, for example "ipfs://CID/123"
- feat: add prop to define how many images to show in the offer full description
- fix: copy assets scripts to copy 1 level of nesting folders
- chore: change tsconfig to allowJs
- feat: add prop to define how many images you want in the offer full description

### CODE REVIEW

File: packages/react-kit/src/components/logo/useAssetLogoSource.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  const networkName = chainId ? chainIdToNetworkName(chainId) : "ethereum";
+  const checksummedAddress = isAddress(address);
+
+  if (checksummedAddress) {
+    return `https://raw.githubusercontent.com/Uniswap/assets/master/blockchains/${networkName}/assets/${checksummedAddress}/logo.png`;
+  } else {
+    return backupImg ?? undefined;
+  }
+}
+const hideLogo = false;
```
[the remark this answers is not in the document]
Comment (albertfolch-redeemeum) [reply packages/react-kit/src/components/logo/useAssetLogoSource.ts]: removed

File: packages/react-kit/src/hooks/contracts/erc1155/useErc1155Uri.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      tokenIds.map((tokenId) =>
+        tokenId
+          ? coreSDK.erc1155Uri({
+              contractAddress,
+              tokenId
+            })
+          : null
+      )
+    );
+  });
```
Comment (levalleux-ludo) [inline packages/react-kit/src/hooks/contracts/erc1155/useErc1155Uri.ts:27]:
```suggestion
  return useQuery(
    ["useErc1155Uri", coreSDK.uuid, props],
    async () => {
      const { contractAddress, tokenIds } = props;
      if (!contractAddress || !tokenIds) {
        return;
      }
      return Promise.all(
        tokenIds.map((tokenId) =>
          tokenId
            ? coreSDK.erc1155Uri({
                contractAddress,
                tokenId
              })
            : null
        )
      );
    },
    { enabled }
  );
```

File: packages/react-kit/src/hooks/contracts/erc721/useErc721TokenUri.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      tokenIds.map((tokenId) =>
+        tokenId
+          ? coreSDK.erc721TokenUri({
+              contractAddress,
+              tokenId
+            })
+          : null
+      )
+    );
+  });
```
Comment (levalleux-ludo) [inline packages/react-kit/src/hooks/contracts/erc721/useErc721TokenUri.ts:27]:
```suggestion
  return useQuery(
    ["useErc721TokenUri", coreSDK.uuid, props],
    async () => {
      const { contractAddress, tokenIds } = props;
      if (!contractAddress || !tokenIds) {
        return;
      }
      return Promise.all(
        tokenIds.map((tokenId) =>
          tokenId
            ? coreSDK.erc721TokenUri({
                contractAddress,
                tokenId
              })
            : null
        )
      );
    },
    { enabled }
  );
```

Comment (levalleux-ludo) [reply packages/react-kit/src/hooks/contracts/erc721/useErc721TokenUri.ts:27]: And the same change is also required in
- packages/react-kit/src/hooks/contracts/erc721/useErc721OwnerOf.ts
- packages/react-kit/src/hooks/offer/useExchangeTokenBalance.ts

**Review by levalleux-ludo — CHANGES_REQUESTED** [review]

**Review by levalleux-ludo — APPROVED** [review]
LGTM

**Pull request merged**